### PR TITLE
Wrap systemd-provided sockets in a TLS listener.

### PR DESCRIPTION
**Pull request by jkroll-cf**
This fixes a bug introduced by the new socket activation/systemd-brokered listening whereby Red October speaks HTTP rather than HTTPS over that socket.

### CODE REVIEW

**Comment by lziest** [comment]
LGTM

File: redoctober.go [hunk 1]
```diff
@@ -134,9 +134,9 @@ func NewServer(process chan<- userRequest, staticPath, addr, caPath string, cert
 			log.Fatal(err)
 		}
 		if len(listenFDs) != 1 {
-			log.Fatal("Unexpected number of socket activation FDs!")
+			log.Fatalf("Unexpected number of socket activation FDs! (%v)", len(listenFDs))
 		}
-		lstnr = listenFDs[0]
+		lstnr = tls.NewListener(listenFDs[0], &config)
```
Comment (jacobhaven) [inline redoctober.go]: I would generally define `config := &tls.Config{...}`

File: redoctober.go [hunk 2]
```diff
@@ -134,9 +134,9 @@ func NewServer(process chan<- userRequest, staticPath, addr, caPath string, cert
 			log.Fatal(err)
 		}
 		if len(listenFDs) != 1 {
-			log.Fatal("Unexpected number of socket activation FDs!")
+			log.Fatalf("Unexpected number of socket activation FDs! (%v)", len(listenFDs))
```
Comment (jacobhaven) [inline redoctober.go]: Might as well call it a `%d`, it's a bit clearer to specify the type if you know it.

Comment (jkroll-cf) [inline redoctober.go]: Addressed Jacob's comments and improved TLS config while I'm in there.

**Comment by jacobhaven** [comment]
LGTM

**Pull request merged**